relational_operators/UnionAllOperator.cpp

```diff
@@ -128,14 +128,56 @@ bool UnionAllOperator::getAllWorkOrders(
 }
 
 bool UnionAllOperator::getAllWorkOrderProtos(WorkOrderProtosContainer* container) {
-  // TODO(tianrun): Add protobuf for UnionAllWorkOrder to support distributed mode.
-  LOG(FATAL) << "UnionAllOperator is not supported in distributed mode yet.";
-  return true;
+  if (!stored_generated_) {
+    for (std::size_t relation_index = 0; relation_index < input_relations_.size(); ++relation_index) {
```

Comment:
I'd suggest to refactor this into a small `generateStored (?)` method, because it seems like isolatable functionality.